jax/experimental/jax2tf/jax2tf.py

```diff
@@ -620,6 +620,62 @@ def dtype_of_val(val: TfVal) -> DType:
   tval, _ = _tfval_to_tensor_jax_dtype(val)
   return tval.dtype
 
+@partial(api_util.api_hook, tag="jax2tf_eval_polymorphic_shapes")
+def eval_polymorphic_shape(fun_jax: Callable,
+               *,
+               polymorphic_shapes=None) -> Callable:
+  """Evaluates the output shape in presence of shape polymorphism.
+
+  This is done without lowering or executing the function, same as for
+  `jax.eval_shape`.
+
+  Args:
+    fun_jax: target JAX function to be called. Its arguments and return value
+      should be JAX arrays, or nested standard Python containers
+      (tuple/list/dict) thereof (pytrees).
+    polymorphic_shapes: Specifies input shapes to be treated polymorphically
+      during shape evaluation. See discussion for `jax2tf.convert`.
+
+      .. warning:: The shape-polymorphic lowering is an experimental feature.
+
+  Returns: a function that takes `jax.ShapeDtypeStruct`s (or any values
+    with `.shape` and `.dtype` attributes) corresponding to the inputs for
+    `fun_jax`, and returns a tuple with:
+
+      * the jax.ShapeDtypeStruct corresponding to the result, as for
+       `jax.eval_shape`. The shape may contain symbolic dimension expressions.
+      * the value that can be passed to `polymorphic_shapes` for a subsequent
+        call to `jax2tf.eval_polymorphic_shape`, or `jax2tf.convert`.
+
+  For example:
+
+  >>> import jax
+  >>> from jax.experimental import jax2tf
+  >>> from jax import numpy as jnp
+  >>>
+  >>> f = lambda A, x: jnp.sin(jnp.dot(A, x))
+  >>> A = jax.ShapeDtypeStruct((2000, 3000), jnp.float32)
+  >>> x = jax.ShapeDtypeStruct((3000, 1000), jnp.float32)
+  >>> out_spec, out_poly_shape = jax2tf.eval_polymorphic_shape(f, polymorphic_shapes=["a, b", "b, c"])(A, x)
```

Comment:
They could be, the parentheses are optional. At this point all of the tests are missing the parentheses because it is more readable that way, so I would leave this as is,